### PR TITLE
Refactor chaos plugin

### PR DESCRIPTION
**Pull request by tyagiparth**

### CODE REVIEW

File: chaos/table_chaos_get_retry_no_config.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		},
+		Columns: []*plugin.Column{
+			{Name: "id", Type: proto.ColumnType_INT},
+			{Name: "retriable_error", Type: proto.ColumnType_STRING},
+		},
+	}
+}
+
+func getRetryNoConfigList(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	log.Println("[INFO] INSIDE LIST CALL")
```
Comment (kaidaguerre) [inline chaos/table_chaos_get_retry_no_config.go:31]: do we need to keep these logs? If you feel they are useful, let's make them lower case

Comment (kaidaguerre) [reply chaos/table_chaos_get_retry_no_config.go:31]: also, make them DEBUG level
```
"[DEBUG] Inside chaos_get_retry_no_config List call")
```

File: chaos/table_chaos_list.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
+package chaos
+
+import (
+	"context"
+	"errors"
+	log "log"
+	"time"
+
+	"github.com/turbot/go-kit/helpers"
+
```
Comment (kaidaguerre) [inline chaos/table_chaos_list.go:10]: remove line

File: chaos/table_chaos_hydrate_retry_no_config.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package chaos
+
+import (
+	"context"
+	"errors"
+	log "log"
```
Comment (kaidaguerre) [inline chaos/table_chaos_hydrate_retry_no_config.go:6]: remove alias

File: chaos/table_chaos_list.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+		return getList(listConfig)(ctx, d, h)
+	}
+	if helpers.StringSliceContains(d.QueryContext.Columns, "panic_after_streaming") {
+		listConfig := &listConfig{listError: FailPanic, rowCount: 15, listErrorRows: 5}
+		return getList(listConfig)(ctx, d, h)
+	}
+	return nil, nil
+}
+
+//// list function ////
```
Comment (kaidaguerre) [inline chaos/table_chaos_list.go:92]: change comment to 
```
// factory function which returns a list call with the behaviour determined by the list config
```

File: chaos/table_chaos_list.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+		if listConfig.listDelay {
+			time.Sleep(delayValue)
+		}
+
+		// rowCount is the number of rows to return
+		rowCount := listConfig.rowCount
+		if rowCount == 0 {
+			rowCount = defaultRowCount
+		}
+		log.Printf("[WARN] THE NUMBER OF ROWS ====>%v", rowCount)
```
Comment (kaidaguerre) [inline chaos/table_chaos_list.go:108]: log needed ?

**Review by kaidaguerre — CHANGES_REQUESTED** [review]
looks good. Logging needs tidying - for each log, consider if it is necessary at all, now things are working. If you decide to keep it, make it DEBUG level an dnicely phrased, including the table name as appropriate

File: chaos/table_chaos_get.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package chaos
```
[the remark this answers is not in the document]
Comment (tyagiparth) [reply chaos/table_chaos_get.go]: It was half done refactoring table for get calls which I'm removing for now. Can add it later when we come to refactoring get calls.

**Pull request merged**